internal/httphandler/data_request_handler.go

```diff
@@ -42,6 +42,13 @@ func NewDataRequestHandler(db bcdb.DB, logger *logger.SugarLogger) http.Handler
 		logger: logger,
 	}
 
+	rangeKeys := []string{
+		"startkey", "{startkey}",
+		"endkey", "{endkey}",
+		"limit", "{limit}",
+	}
```

Comment:
This assumes that all 3 parameters must exist, otherwise there is a 404, which is difficult to understand. This might be improved to be more flexible, by adding matchers to queries with default parameters and setting defaults. 
Or, add a "missing parameter" matcher that would give an informative response... e.g. `missing startkey, endkey, or limit` 

It is a bit weird to do `xxx?startkey=&endkey=&limit=0` in order to get the entire range no?
Or maybe it doesn't matter because it is all produced by the SDK?

Reply:
> Or maybe it doesn't matter because it is all produced by the SDK?

Yes. However, we do support cURL based usage too. Hence, we can add that in the subsequent PRs.